packages/hdwallet-provider/src/index.ts

```diff
@@ -271,21 +276,36 @@ class HDWalletProvider {
         // @ts-ignore - the type doesn't take into account the possibility
         // that response.error could be a thing
         (error: any, response: JSONRPCResponsePayload & { error?: any }) => {
+          // always resolve the Promise - if an error occurs, throw it when
+          // this.initialized is checked so errors can be handled by the user
           if (error) {
-            reject(error);
+            resolve({
+              success: false,
+              error
+            });
             return;
           } else if (response.error) {
-            reject(response.error);
+            resolve({
+              success: false,
+              error: response.error
+            });
             return;
           }
```

Comment:
This is a minor style nit that you are free to ignore... the code could be written as
```javascript
if (error) {
  return resolve({});
}

if (response.error) {
  return resolve({})
}

// process response

Reply:
Yeah that looks a lot nicer. I was avoiding this because I actually didn't want to return anything explicitly. But this just seems a lot cleaner.